### PR TITLE
refactor: migrate mobile number validator to ts

### PR DESCRIPTION
**Pull request by tshuli**
## Problem
Part of #7

## Solution

**Improvements**

- Migrates Mobile Number validation to Typescript

## Tests
- [ ] Create two mobile phone number fields. Hide one of them behind logic.
- [ ] Check that validation on both visible and hidden phone number fields work when submitting programatically
 - Invalid phone number format is rejected
 - Valid phone number format is accepted
 - When international number are allowed, international phone number is accepted
 - When international number is not allowed, international phone number is rejected

### CODE REVIEW

File: src/app/utils/field-validation/validators/mobileNoValidator.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+const mobilePhoneNumberValidator: MobileNoValidatorConstructor = () => (
+  response,
+) => {
+  return isMobilePhoneNumber(response.answer)
+    ? right(response)
+    : left(`MobileNoValidator:\t answer is not a valid mobile phone number`)
+}
+
+const prefixValidator: MobileNoValidatorConstructor = (homeNumberField) => (
```
Comment (karrui) [inline src/app/utils/field-validation/validators/mobileNoValidator.ts:28]: should be `mobileNumberField` as parameter?

File: src/app/utils/field-validation/validators/mobileNoValidator.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  response,
+) => {
+  if (
```
Comment (karrui) [inline src/app/utils/field-validation/validators/mobileNoValidator.ts:31]: nit: return ternary for easier parsing? and reverse the order so we don't need to read so many negations

```ts
return (mobileNoField.allowIntlNumbers || startsWithSgPrefix(response.answer))
  ? right(response)
  : left(`MobileNoValidator:\t answer is not an SG number but intl numbers are not allowed`)
```

[de morgan's law](https://csawesome.runestone.academy/runestone/books/published/csawesome/Unit3-If-Statements/topic-3-6-DeMorgan.html) btw:
current: `!a && !b`
to reverse order:
`!(!a && !b) === !!a || !!b === a || b`

**Review by karrui — APPROVED** [review]
lgtm, some nits

File: src/app/utils/field-validation/validators/mobileNoValidator.ts (hunk 2 above)
Comment (tshuli) [reply src/app/utils/field-validation/validators/mobileNoValidator.ts]: cool stuff, thanks for pointing out!

**Pull request merged**